src/Examples/VGG.cs

```diff
@@ -0,0 +1,211 @@
+                    }
+
+                    batchId++;
+
+                    data.Dispose();
+                    target.Dispose();
+
+                }
+            }
+
+        }
+
+        /// <summary>
+        ///     
+        /// </summary>
+        private class Model : Module
+        {
+            Sequential sequential;
+            Module classifier;
+            private readonly static Dictionary<string, object[]> cfgs = new Dictionary<string, object[]>();
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="VGGName"></param>
+            public Model(string VGGName) : base()
+            {
+                object[] vgg11 = new object[] { 64, 'M', 128, 'M', 256, 256, 'M', 512, 512, 'M', 512, 512, 'M' };
+                object[] vgg13 = new object[] { 64, 64, 'M', 128, 128, 'M', 256, 256, 'M', 512, 512, 'M', 512, 512, 'M' };
+                object[] vgg16 = new object[] { 64, 64, 'M', 128, 128, 'M', 256, 256, 256, 'M', 512, 512, 512, 'M', 512, 512, 512, 'M' };
+                object[] vgg19 = new object[] { 64, 64, 'M', 128, 128, 'M', 256, 256, 256, 256, 'M', 512, 512, 512, 512, 'M', 512, 512, 512, 512, 'M' };
+
+                cfgs.Add("VGG11", vgg11);
+                cfgs.Add("VGG13", vgg13);
+                cfgs.Add("VGG16", vgg16);
+                cfgs.Add("VGG19", vgg19);
+
+                object[] cfg = cfgs.GetValueOrDefault(VGGName);
+                this.sequential = MakeSequential(cfg);
+                classifier = Linear(320, 50);
+
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="input"></param>
+            /// <returns></returns>
+            public override TorchTensor Forward(TorchTensor input)
+            {
+                TorchTensor res = this.sequential.Forward(input);
+                res = res.View(new long[] { -1, 0 });
+                return classifier.Forward(res);
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="cfg"></param>
+            /// <returns></returns>
+            private Sequential MakeSequential(object[] cfg)
+            {
+                List<Module> layers = new List<Module>();
+                int in_channels = 3;
+
+                foreach (object x in cfg)
+                {
+                    if (x.ToString() == "M")
+                        layers.Add(Module.MaxPool2D(new long[2], new long[1]));
+                    else
+                    {
+                        layers.Add(Module.Conv2D(in_channels, (long)x, 3, padding: 1));
+                        //layers.Add(Module.Bat);
+                        layers.Add(Module.Relu(inPlace: true));
+                        in_channels = (int)x;
+                    }
+                }
+                // layers.Add(Module.AdaptiveAvgPool2D();
```

Comment:
We also have avg pool, so please add this as well.

Reply:
agree, I have added it, also I had to create a override function for avgpool2d which i have added with this PR